frontend/src/components/VulnerableGpaComponent.tsx

```
@@ -0,0 +1,238 @@
+    onDataSubmit 
+}) => {
+    const [htmlContent, setHtmlContent] = useState('');
+    const [userCode, setUserCode] = useState('');
+    const [redirectUrl, setRedirectUrl] = useState('');
+    const iframeRef = useRef<HTMLIFrameElement>(null);
+
+    const renderUserHtml = (html: string) => {
+        return (
+            <div 
+                style={{ border: '1px solid #ccc', padding: '10px', margin: '10px' }}
+                dangerouslySetInnerHTML={{ __html: html }}
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)